starknet_devnet/starknet_wrapper.py

```diff
@@ -564,12 +564,11 @@ async def calculate_actual_fee(self, external_tx: InvokeFunction):
             signature=external_tx.signature,
             nonce=None,
             chain_id=state.general_config.chain_id.value,
-            # Need to set to 0 as it will be invoked in apply_state_updates
-            version=constants.TRANSACTION_VERSION,
+            version=external_tx.version or constants.QUERY_VERSION,
```

Comment:
#hacky
Can you add a #comment describing why it works so that we know in future.

Reply:
How about this? [e801652](https://github.com/Shard-Labs/starknet-devnet/pull/88/commits/e80165286b46227410f8f90e7eaf77a36c1ccc0e)

Reply:
very nice, great success